src/traces/scattergl/index.js

```diff
@@ -395,15 +395,16 @@ function plot(gd, subplot, cdata) {
             scene.line2d.update(scene.lineOptions);
             scene.lineOptions = scene.lineOptions.map(function(lineOptions) {
                 if(lineOptions && lineOptions.positions) {
-                    var pos = [], srcPos = lineOptions.positions;
+                    var pos = [];
+                    var srcPos = lineOptions.positions;
 
                     var firstptdef = 0;
-                    while(isNaN(srcPos[firstptdef]) || isNaN(srcPos[firstptdef + 1])) {
+                    while(firstptdef < srcPos.length && (isNaN(srcPos[firstptdef]) || isNaN(srcPos[firstptdef + 1]))) {
                         firstptdef += 2;
                     }
                     var lastptdef = srcPos.length - 2;
-                    while(isNaN(srcPos[lastptdef]) || isNaN(srcPos[lastptdef + 1])) {
-                        lastptdef += -2;
+                    while(lastptdef > -1 && (isNaN(srcPos[lastptdef]) || isNaN(srcPos[lastptdef + 1]))) {
```

Comment:
even better: `while(lastptdef > firstptdef && ...` would skip this second loop altogether in the failing case.

BTW do we need this `pos = pos.concat`... thing at all? Can't we just do `lineOptions.positions = srcPos.slice(...)`? It's needed in the fill section below but I don't see why it'd be needed here. In fact, even that `slice` seems like it could be omitted when we didn't drop any points on either end.